src/components/editor/useSupportedLanguages.ts

```diff
@@ -3,22 +3,21 @@ import {
   useState,
 } from 'react';
 
-import { languages } from 'monaco-editor';
+import { useMonaco } from '@monaco-editor/react';
 
 export const useSupportedLanguages = (): string[] => {
   const [currentLanguages, setCurrentLanguages] = useState<string[]>([]);
-  // Get the supported
+  const monaco = useMonaco();
```

Comment:
Does this create an extra instantiation of the editor?

Reply:
It doesn't create an extra, it should just return the current reference. The other way to do this is by using a ref and getting the reference after the component mounts.